File: src/community/script/py/src/test/resources/org/geoserver/script/py/tx.py
from geoserver.wfs import tx 

@tx.before
def onBefore(req, context):
  context['before'] = True

@tx.preInsert
def onPreInsert(inserted, req, context):
  context['preInsert'] = True

@tx.postInsert
def onPostInsert(inserted, req, context):
  context['postInsert'] = True

@tx.preUpdate
def onPreUpdate(updated, props, req, context):
  context['preUpdate'] = True

@tx.postUpdate
def onPostUpdate(updated, props, req, context):
  context['postUpdate'] = True

@tx.preDelete
def onPreDelete(deleted, req, context):
  context['preDelete'] = True

@tx.postDelete
def onPostDelete(deleted, req, context):
  context['postDelete'] = True

@tx.preCommit
def onPreCommit(req, context):
  context['preCommit'] = True

@tx.postCommit  
def onPostCommit(req, res, context):
  context['postCommit'] = True

@tx.abort
def onAbort(req, res, context):
  context['abort'] = True
 
  

File: src/community/script/rb/src/test/resources/org/geoserver/script/rb/tx.rb
def before(req, context)
  context["before"] = true;
end

def preInsert(inserted, req, context)
  context["preInsert"] = true;
end

def postInsert(inserted, req, context)
  context["postInsert"] = true;
end

def preUpdate(updated, props, req, context)
  context["preUpdate"] = true;
end

def postUpdate(inserted, props, req, context)
  context["postUpdate"] = true;
end

def preDelete(deleted, req, context)
  context["preDelete"] = true;
end

def postDelete(deleted, req, context)
  context["postDelete"] = true;
end

def preCommit(req, context)
  context["preCommit"] = true;
end

def postCommit(req, res, context)
  context["postCommit"] = true;
end

def abort(req, res, context)
  context["abort"] = true;
end
